packages/cart/src/services/cart-module.ts

```diff
@@ -118,29 +127,34 @@ export default class CartModuleService implements ICartModuleService {
       sharedContext
     )
 
-    return (Array.isArray(data) ? result : result[0]) as CartDTO | CartDTO[]
+    return (Array.isArray(data) ? result : result[0]) as
+      | CartTypes.CartDTO
+      | CartTypes.CartDTO[]
   }
 
   @InjectTransactionManager("baseRepository_")
   protected async create_(
-    data: CreateCartDTO[],
+    data: CartTypes.CreateCartDTO[],
     @MedusaContext() sharedContext: Context = {}
   ) {
     return await this.cartService_.create(data, sharedContext)
```

Comment:
suggestion: in the case the protected method only does a call to the underlyin service, should we remove the protected method and move that to the public one + InjectTransactionManager. Or is that in case we need to add anything in the future?

Reply:
Yes, I used this pattern because I am almost 100% certain we will need update this method in the near future